src/mixins/filing-template-mixin.ts

```diff
@@ -187,21 +186,17 @@ export default class FilingTemplateMixin extends Vue {
   /**
    * Prepare name translations for non draft correction
    */
-  prepareNameTranslations () : NameTranslationIF {
-    const translations = this.getNameTranslations as NameTranslationDraftIF[]
-    return {
-      new: translations?.filter(x => x.action === ActionTypes.ADDED)
-        .map(x => x.value),
-      modified: translations?.filter(x => x.action === ActionTypes.EDITED)
-        .map(x => {
-          return {
-            newValue: x.value,
-            oldValue: x.oldValue
-          }
-        }),
-      ceased: translations?.filter(x => x.action === ActionTypes.REMOVED)
-        .map(x => x.value)
-    } as NameTranslationIF
+  prepareNameTranslations () : NameTranslationIF[] {
+    return this.getNameTranslations?.filter(x => x.action === ActionTypes.ADDED || x.action === ActionTypes.EDITED)
+      .map(x => {
+        let nameTranslation = {
+          name: x.name
+        }
+        if (x.action === ActionTypes.EDITED) {
+          nameTranslation['id'] = x.id
```

Comment:
I didn't see where "id" was originally assigned...?

If it orginally comes from the API, how does it get updated for new names?

Reply:
Based on schema id should not be included for new translation

Reply:
OK. So what is ID used for?

Reply:
for modified translation

Reply:
And where is it assigned?

Reply:
line no 217

Reply:
```
id: x.id,
```
But where did `x.id` come from? I don't see where id is assigned a number.

Reply:
pk from database. FE wont be setting any value for id

Reply:
Oh OK. So the UI maintains it if the API provided it, otherwise it is empty?

Reply:
Yes. API does provide ID for the translations which is already been part of IA